kore/src/Kore/Equation/DebugEquation.hs

```diff
@@ -276,6 +272,39 @@ instance Entry DebugAttemptEquation where
     oneLineDoc (DebugAttemptEquationResult _ (Left _)) = "equation is not applicable"
     oneLineDoc (DebugAttemptEquationResult _ (Right _)) = "equation is applicable"
 
+    oneLineJson = \case
+        entry@(DebugAttemptEquation equation _) ->
+            JSON.object
+                [ "tag" JSON..= entryTypeText (toEntry entry)
+                , "rule-id"
+                    JSON..= fromMaybe "UNKNOWN" (Attribute.getUniqueId . Attribute.uniqueId . attributes $ equation)
+                ]
+        _entry@(DebugAttemptEquationResult equation result) ->
+            let resultInfo = case result of
+                    Right _ ->
+                        JSON.object
+                            [ "tag" JSON..= ("success" :: Text)
+                            , "rule-id" JSON..= ruleIdText
+                            ]
+                    Left failure ->
+                        JSON.object
+                            [ "tag" JSON..= ("failure" :: Text)
+                            , "rule-id" JSON..= ruleIdText
+                            , "reason" JSON..= failureDescription failure
+                            ]
+             in JSON.object
+                    [ "tag" JSON..= ("simplification" :: Text)
+                    , "result" JSON..= resultInfo
+                    ]
+          where
+            failureDescription :: AttemptEquationError RewritingVariableName -> Text
+            failureDescription err = shortenText . Pretty.renderText . Pretty.layoutOneLine . Pretty.pretty $ err
+
+            shortenText :: Text -> Text
+            shortenText msg = Text.take 200 msg <> ("...truncated" :: Text)
```

Comment:
should we have this here? I notice that we still emit stuff like `/* T Fn D Spa */` in kore terms. Perhaps once those are removed, we can remove this truncation without things blowing up too much?

Reply:
I added the truncation because the golden file got to 140mb... though that was a bit too much to put into git.

Reply:
https://github.com/runtimeverification/haskell-backend/pull/3823 did not improve matters much unfortunately. I've settled on a slightly longer cutoff of 500 characters.